src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -623,8 +624,10 @@ internal Boolean LoadLocalHeaderExtraFieldAndCompressedBytesIfNeeded()
                 _compressedBytes = new Byte[(_compressedSize / MaxSingleBufferSize) + 1][];
                 for (int i = 0; i < _compressedBytes.Length - 1; i++)
                 {
-                    _compressedBytes[i] = new Byte[MaxSingleBufferSize];
+                    _compressedBytes[i] = ArrayPool<byte>.Shared.Rent(MaxSingleBufferSize);
```

Comment:
Yeah, we would need to add Dispose and the compression components would have to call it.